src/rtc.rs

```diff
@@ -416,38 +414,21 @@ fn hours_to_u8(hours: Hours) -> Result<u8, Error> {
 /// Enable the low frequency external oscillator. This is the only mode currently
 /// supported, to avoid exposing the `CR` and `CRS` registers.
 fn enable_lse(rcc: &RegisterBlock, bypass: bool) {
-    unsafe {
-        // Force a reset of the backup domain.
-        // Set BDCR - Bit 16 (BDRST)
-        bb::set(&rcc.bdcr, 16);
-        // Clear BDCR - Bit 16 (BDRST)
-        bb::clear(&rcc.bdcr, 16);
-        // Enable the LSE.
-        // Set BDCR - Bit 0 (LSEON)
-        bb::set(&rcc.bdcr, 0);
-        if bypass {
-            // Set BDCR - Bit 2 (LSEBYP)
-            bb::set(&rcc.bdcr, 2);
-        } else {
-            // Clear BDCR - Bit 2 (LSEBYP)
-            bb::clear(&rcc.bdcr, 2);
-        }
-        while rcc.bdcr.read().lserdy().bit_is_clear() {}
-        // Set clock source to LSE.
-        // Set BDCR - Bit 8 (RTCSEL to value for LSE)
-        bb::set(&rcc.bdcr, 8);
-    }
+    // Force a reset of the backup domain.
+    rcc.bdcr.modify(|_, w| w.bdrst().set_bit());
+    rcc.bdcr.modify(|_, w| w.bdrst().clear_bit());
+    // Enable the LSE.
+    rcc.bdcr
+        .modify(|_, w| w.lseon().set_bit().lsebyp().bit(bypass));
+    while rcc.bdcr.read().lserdy().bit_is_clear() {}
+    // Set clock source to LSE.
+    rcc.bdcr.modify(|_, w| w.rtcsel().lse());
```

Comment:
This was also using `bb` to avoid races, that `&RegisterBlock` is acquired with `unsafe`